main.go

```diff
@@ -72,8 +73,22 @@ const (
 	AccountEmpty
 )
 
+// Result is the action taken by a swipe of a card. It contains all information
+// to be communicated to the user.
+type Result struct {
+	Code    ResultCode
+	UID     []byte
+	User    string
+	Account float32
```

Comment:
You can dis-/charge arbitrary amounts (e.g. someone takes pizza-money out of their account), so this can get fractional. It's normally an integer, but for printing (which is the only thing this is used for) this is simpler.